src/PhpSpreadsheet/Shared/CodePage.php

```diff
@@ -82,8 +83,19 @@ public static function validate(string $codePage): bool
      */
     public static function numberToName(int $codePage): string
     {
-        if (array_key_exists($codePage, self::$pageArray)) {
-            return self::$pageArray[$codePage];
+        if (array_key_exists($codePage, self::PAGE_ARRAY)) {
+            $value = self::PAGE_ARRAY[$codePage];
+            if (is_array($value)) {
+                foreach ($value as $encoding) {
+                    if (@iconv('UTF-8', $encoding, ' ') !== false) {
```

Comment:
Is there any way that we could identify which of the possible charsets for a codepage value is being used on the first request for a file, and then cache that result so that we don't have to do the check for every single request? It's not critical, because it's never going to be more than 3 requests per file, but just wondering.

Reply:
Done. Tradeoff is making array static rather than const.